### ISSUE TITLE
Not working on Amazon Prime video

### ISSUE THREAD

**Issue by dorkirt** [muzuiget/dualsub-support#151]
Not working on Amazon Prime video, subtitles are totally out of sync.

**Comment by muzuiget** [comment]
Please provide the video url. Or can you find out really sync offset?

Currently, some videos the time-point in the video and int the subtitle are not sync, it has 10s delay, I am not sure all videos are not sync.

Here is my [test video](https://www.primevideo.com/detail/0T1R1MXA75ZCR0C3247CHPIS3X/ref=atv_tv_hom_c_I4XQfs_3_4)

Are the out of sync on your computer?

**Comment by dorkirt** [comment]
Strange, i have that problem in every video of primevideo...even on your test video 1x01. I was trying

https://www.primevideo.com/detail/0IIETVO97D3VQICQYX8LTIDPL4/ref=atv_dp_season_select_s3

I said there was a sync problem, but the true is the subtitle doesnt seems to fit on that chapter... Netflix works fine for me though.

**Comment by muzuiget** [comment]
Because only on primevideo, the subtitle time-points not sync with the video, maybe the primevideo player has a "hidden" value so it can knows the correct offset and syncs the subtitle with the video.

I am not sure this "offset time" is random or fixed on difference videos. If you want to help debugger, here is the way.

Show the primevideo native subtitle and the Dualsub subtitle in the same time, so you can check the Dualsub subtitle is earlier or later than the the native subtitle(the current time). 

Test with this way in difference videos, find out the offset time is random or fixed.

If it is random, I need to take more time to find where to get the correct offset.

**Comment by dorkirt** [comment]
The subtitles on 3x14 (at 1:18) are from 3x15 (at 1:08)

https://www.primevideo.com/detail/0IIETVO97D3VQICQYX8LTIDPL4/ref=atv_dp_season_select_s3

**Comment by muzuiget** [comment]
Currently my primevideo account expired. This bug may be fixed later.

**Comment by muzuiget** [comment]
The 3x14 (at 1:18) is find on my browser.

![www primevideo com_region_fe_detail_0IIETVO97D3VQICQYX8LTIDPL4_ref=atv_dp_season_select_s3](https://user-images.githubusercontent.com/330812/96413660-8be08c80-121e-11eb-9551-114e83890d52.jpg)

**Comment by muzuiget** [comment]
All right, it look like I found the reason: difference OS browser has difference behavior, I can reproduce your issue after changing the browser `user-agent` string. 

I develop the extension on Chrome for Linux, Dualsub always use the last subtitle which it captured.

But on Chrome for Windows, the website will *pre-load* next episode's subtitle, so that is why you playing 3x14, but the subtitle is 3x15.

Worse, if you jump to next episode, you will notice the url in the browser location bar won't changed, so Dualsub can't get the correct subtitle should use.

A workaround: you can change your `user-agent` to Chrome for Linux, here is mine:

```
Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.75 Safari/537.36
```

**Comment by muzuiget** [comment]
Document updated: https://www.dualsub.xyz/website.html

Amazon Prime Video classified as Level C.

**Comment by muzuiget** [comment]
This bug finally fixed in v1.50.2
